pkg/ansible/proxy/proxy.go

```diff
@@ -189,12 +188,36 @@ func InjectOwnerReferenceHandler(h http.Handler, cMap *ControllerMap) http.Handl
 			log.V(1).Info("serialized body", "Body", string(newBody))
 			req.Body = ioutil.NopCloser(bytes.NewBuffer(newBody))
 			req.ContentLength = int64(len(newBody))
-			// add watch for resource
-			err = addWatchToController(owner, cMap, data)
+			dataMapping, err := restMapper.RESTMapping(data.GroupVersionKind().GroupKind(), data.GroupVersionKind().Version)
 			if err != nil {
-				m := "could not add watch to controller"
+				m := fmt.Sprintf("could not get rest mapping for: %v", data.GroupVersionKind())
 				log.Error(err, m)
 				http.Error(w, m, http.StatusInternalServerError)
+				return
+			}
+			// We need to determine whether or not the owner is a cluster-scoped
+			// resource because enqueue based on an owner reference does not work if
+			// a namespaced resource owns a cluster-scoped resource
+			ownerGV, err := schema.ParseGroupVersion(owner.APIVersion)
+			ownerMapping, err := restMapper.RESTMapping(schema.GroupKind{Kind: owner.Kind, Group: ownerGV.Group}, ownerGV.Version)
+			if err != nil {
+				m := fmt.Sprintf("could not get rest mapping for: %v", owner)
+				log.Error(err, m)
+				http.Error(w, m, http.StatusInternalServerError)
+				return
+			}
+
+			dataClusterScoped := dataMapping.Scope.Name() != meta.RESTScopeNameRoot
+			ownerClusterScoped := ownerMapping.Scope.Name() != meta.RESTScopeNameRoot
```

Comment:
I may be misinterpreting, but `meta.RESTScopeNameRoot` is the name for the cluster scope right (as opposed to `meta.RESTScopeNameNamespace`)? Shouldn't this be `ownerClusterScoped := ownerMapping.Scope.Name() == meta.RESTScopeNameRoot` (i.e. `==`, not `!=`). Same for `dataClusterScoped`.

If so, does that change the logic of the if statement in line 212?